snyk/policy
# feat: support security policies ignore action

#### What does this PR do?

Implementing new source of ignore data from security policies.
In case issue was enriched with `securityPolicyMetaData.ignore` this issue should apply ignore from security policy only and skip other ignores. Also ignore object will contain few more fields and will look like:

```js
'npm:tar:20151103': [
  {
    reason: '',
    reasonType: 'wont-fix',
    source: 'security-policy',
    ignoredBy: {
      id: '22A6B3BE-ABEF-4407-A634-AB1BE30A552F',
      name: 'Ignored by Security Policy',
    },
    created: '2021-06-13T09:33:57.318Z',
    disregardIfFixable: false,
    path: ['*'],
  },
]
```

In other case old logic should be applied.

#### How should this be manually tested?

1. Checkout this branch locally and use it as local source for `snyk-policy` in registry
2. Enable ignore action FF in registry
3. Create new policy with the ignore action
4. Re-test project from the org where you attached new policy
5. You should see issues that match policy ignored

**Alternatively to step 1** you can open `node_modules/snyk-policy/lib/filter/ignore.js` and replace it with the contents of same file from this PR. And restart registry
